### PR TITLE
Transaction id should be optional for correlation info

### PR DESCRIPTION
**Pull request by tomkerkhove**
Transaction id should be optional for correlation info.

In a Web API context we'll generate one, but messaging just uses what's there so this should be optional.

### CODE REVIEW

File: src/Arcus.Observability.Correlation/CorrelationInfo.cs [hunk 1]
```diff
@@ -15,9 +14,6 @@ public CorrelationInfo(string operationId, string transactionId)
         {
             Guard.NotNullOrEmpty(operationId, nameof(operationId),
                 "Cannot create a correlation instance with a blank operation ID");
-            Guard.For<ArgumentException>(
```
Comment (stijnmoreels) [inline src/Arcus.Observability.Correlation/CorrelationInfo.cs]: Ok, optional; but do we allow empty? Shouldn't we allow either filled or `null`?

Comment (tomkerkhove) [reply src/Arcus.Observability.Correlation/CorrelationInfo.cs]: Frankly, we should just accept whatever there is - `null`, ``, ` ` or `abc`.

We don't care, do we?

Comment (stijnmoreels) [reply src/Arcus.Observability.Correlation/CorrelationInfo.cs]: It's discussable, and a bit political 😄 , if it's optional or if it's allowed to be empty. You see. 'empty' can be both seen as input or as something that's between non-input and input.
Hence, 'optional' or 'empty'.

Comment (stijnmoreels) [reply src/Arcus.Observability.Correlation/CorrelationInfo.cs]: I guess it depends on the type and the validation of the model.

Comment (tomkerkhove) [reply src/Arcus.Observability.Correlation/CorrelationInfo.cs]: Let me rephrase - We shouldn't care what the value is, it's up to the user to decide if it's valid or not ;)

Comment (stijnmoreels) [reply src/Arcus.Observability.Correlation/CorrelationInfo.cs]: Ah, yes, ok. Shame though that the validation is not done at creation-level. But, yeah. I see your point.

Comment (stijnmoreels) [reply src/Arcus.Observability.Correlation/CorrelationInfo.cs]: So if it's not `null`, then we have a transaction ID?

Comment (tomkerkhove) [reply src/Arcus.Observability.Correlation/CorrelationInfo.cs]: It's something the consumer has to define if it's valid or not so we can't provide.

Frankly, I don't care :D We just pass it, rest is up to them.

Comment (tomkerkhove) [reply src/Arcus.Observability.Correlation/CorrelationInfo.cs]: It's a functional thing, not a technical one
